lib/dicom_net/association.ex

```diff
@@ -155,19 +158,49 @@ defmodule DicomNet.Association do
   end
 
   defp handle_pdu(
-         %Pdu{type: :data, data: %{pdv_flags: :command_last_fragment, data: data}},
+         %Pdu{
+           type: :data,
+           data: %{
+             pdv_flags: :command_last_fragment,
+             pdv_transfer_ctx_id: presentation_context_id,
+             data: data
+           }
+         },
          %{state: :association_established} = state
        ) do
     Logger.debug("Received command")
 
     command_ds = Dicom.BinaryFormat.from_binary(data, endianness: :little, explicit: false)
+    asci_de = DataSet.value_for!(command_ds, :AffectedSOPClassUID)
+    mid_de = DataSet.value_for!(command_ds, :MessageID)
+
+    case asci_de do
+      @verification_sopclassuid ->
```

Comment:
I think the "original" way would be to check that `command_field == 0x0030` here. I would recommend you to do that, since my gut feeling is, that this is the most robust version against different SCU implementations. That being said, I guess in this particular case, checking the SOP class should also work.

Reply:
Yes, I agree. Replaced.